Lib/test/test_poll.py

```diff
@@ -204,6 +204,39 @@ def test_threaded_poll(self):
             os.write(w, b'spam')
             t.join()
 
+    @unittest.skipUnless(threading, 'Threading required for this test.')
+    @reap_threads
+    def test_poll_blocks_with_negative_ms(self):
+
+        for timeout_ms in [None, -1, -1.0, -0.1, -1e-100]:
+
+            # GIVEN
+
+            # Create two file descriptors. This will be used to unlock
+            # the blocking call to poll.poll inside the thread
+
+            r, w = os.pipe()
```

Comment:
A pair of descriptors is created for every `timeout_ms` value, but they are closed only after finishing the test.

Reply:
Done in 5c58695